File: killrvideo-service-suggestedvideo/src/main/java/com/killrvideo/service/sugestedvideo/dao/VideoMapper.java
package com.killrvideo.service.sugestedvideo.dao;

import com.datastax.oss.driver.api.core.CqlSession;
import com.datastax.oss.driver.api.mapper.MapperBuilder;
import com.datastax.oss.driver.api.mapper.annotations.DaoFactory;
import com.datastax.oss.driver.api.mapper.annotations.Mapper;

@Mapper
public interface VideoMapper {
    @DaoFactory
    VideoDao getVideoDao();

    static MapperBuilder<VideoMapper> build(CqlSession session) {
        return new VideoMapperBuilder(session);
    }
}

File: killrvideo-service-suggestedvideo/src/main/kotlin/com/killrvideo/service/suggestedvideo/dao/VideoMapper.kt
package com.killrvideo.service.suggestedvideo.dao

import com.datastax.oss.driver.api.mapper.annotations.DaoFactory
import com.datastax.oss.driver.api.mapper.annotations.Mapper

@Mapper
interface VideoMapper {
    @get:DaoFactory
    val videoDao: VideoDao
}
